src/functions.ts

```diff
@@ -0,0 +1,112 @@
+import { FINAL_COMMA, FINAL_PAREN, PUNCTUATION } from 'consts'
+import { titleCase, swapCase, upperCaseFirst, capitalCase } from 'text-case'
+
+/**
+ * Functions get passed a mutable html anchor element and the current text
+ * They must return a new text that can be propagated to the next function
+ */
+export interface PipeFunction {
+  (link: HTMLAnchorElement, text: string): string
+}
+
+export const DEFAULT_FUNCTIONS: Array<PipeFunction> = [removeParenthesisComma, blendSuffix]
+
+export const FUNCTION_LOOKUP: Record<string, PipeFunction> = {
+  lower: lower,
+  _: lower,
+
+  upperFirst: upperFirst,
+  upper_first: upperFirst,
+  upperfirst: upperFirst,
+  '^': upperFirst,
+
+  capital: capital,
+  '^^': capital,
```

Comment:
I'm not sure about the aliases for the case transforms. Maybe we can keep the `_` and `^` for now, and figure out whether there's a need for more aliases (and what they should be, later on?

Reply:
See discussion above